app/models/arclight/parents.rb

```diff
@@ -19,7 +20,9 @@ def eadid
     ##
     # @return [Array[Arclight::Parent]]
     def as_parents
-      Hash[ids.zip(labels)].map { |k, v| Arclight::Parent.new(id: k, label: v, eadid: eadid) }
+      parents = []
+      ids.each.with_index { |_id, idx| parents << Arclight::Parent.new(id: ids[idx], label: labels[idx], eadid: eadid, level: levels[idx]) }
```

Comment:
Instead of incrementally building up the `parents` array, could we use `map` instead of `each`?

Reply:
I wonder, too, if there's an opportunity to represent the ids, labels, and levels as an object that we can iterate over?  Perhaps `Parent` is that object?

Reply:
Good idea. I'm changing this is map.with_index.